tests/cri-containerd: split between cri-containerd and crictl suites

wainersm opened the pull request:
This split the cri-containerd in two suites of tests:

* cri-containerd:  keep running containerd's [cri-integration tests](https://github.com/containerd/containerd/blob/main/docs/cri/testing.md#cri-integration-test)
* crictl: new home for the crictl-based tests 

Then the crictl tests were convert to use the bats framework. I didn't  want to improve the tests, depict many opportunities, so I tried to keep the original code and logic intact as much as possible. In the future, when we start pulling tests from CCv0 I think it will be the right time for improvements.

A github workflow was created to run the crictl suite of tests. The workflow won't fail due to failures on tests, because I will be able to fully test the tests on CI after the workflow is merged (right?) and then I don't want a broken workflow blocking unrelated pull requests.

Fixes #8160

wainersm commented:
Hi @fidencio !

> @wainersm, I don't understand [9c1a213](https://github.com/kata-containers/kata-containers/commit/9c1a2132f31c631002b788b5c162cf89b250a179)
> 
> What exactly are you trying to gain with that? integration/cri-containerd should be clear that this is cri-containerd integration tests ... making it integration/cri-containerd-integration is redundant.

Once upon a time PR #8061, that @beraldoleal asked me to help understand why some CI jobs failed. I took look at https://github.com/kata-containers/kata-containers/actions/runs/6356069723/job/17266102296?pr=8061 first, when I realized that our `cri-containerd` suite of tests is actually running two types of tests:

 1. The containerd's cri-integration suite. External to kata's code base, writing in Golang...etc
 2. Custom shell script tests within kata's code base, using crictl/cri/containerd

So  my goal is to split those two types of tests in different suites to easy the interpretation of the logs when the CI fail. The tests 2) (`cri-containerd`) could even growth in the future with the addition of https://github.com/kata-containers/tests/tree/CCv0/integration/containerd/confidential tests.

Let me know if I'm missing something and/or this is an unwanted/worthless split.

fidencio commented:
Aha, okay, it makes sense.

I'd just not name it as cri-containerd-integration :-)

wainersm commented:
> Aha, okay, it makes sense.
> 
> I'd just not name it as cri-containerd-integration :-)

hmm.. what do you suggest? Keep containerd's cri-integration tests called `cri-containerd` and naming our custom tests something else (e.g. `crictl-containerd`) ? Naming is an art my friend...

wainersm commented:
**run-cri-containerd (lts, qemu)** is also failing locally and I still don´t know why.

wainersm commented:
> **run-cri-containerd (lts, qemu)** is also failing locally and I still don´t know why.

Fixed. Added an instruction to pull the pause image before creating the pod.

wainersm commented:
Re-done my initial PR to move the crictl tests to its own suite (guess what, named `crictl`). So the cri-containerd suite is kept running the containerd's cri-integration tests.

Once CI passing I should be working on:
* convert crictl to use bats
* create a new CI job to run crictl only

wainersm commented:
@fidencio this PR is ready to review. The three commits that convert to bats I should squash after the reviews (keeping them separated to ease the reviewer's life).
                                                                     / ---- \
                                                                    / ------ \
@stevenhorsman tentative to build a  |  []  |  (house) for the crictl tests that we will (or not) be pulling to main...

wainersm commented:
> @fidencio this PR is ready to review. The three commits that convert to bats I should squash after the reviews (keeping them separated to ease the reviewer's life). / ---- / ------ @stevenhorsman tentative to build a | [] | (house) for the crictl tests that we will (or not) be pulling to main...

yeah, my tentative to build a house in ascii didn't work out....

stevenhorsman commented:
/ ---- \
    / ------ \
      | [] |

stevenhorsman commented on tests/integration/crictl/lib.sh:
Do we need a copyright statement here (presumably copied from the source of this code)?

stevenhorsman reviewed (approved):
Looks good to me, one minor comment...

wainersm commented:
Rebase this to main, added a header to `tests/integration/crictl/lib.sh` and squashed the commits that convert tests to bats.

stevenhorsman commented:
@wainersm - Do we still want to get this in? I think it needs a rebase and re-test and then we can have another attempt?
